test/Dialect/Torch/scalarize-shapes.mlir

```diff
@@ -374,3 +374,172 @@ func.func @squeeze_dim_full_fold(%arg0: !torch.vtensor<[?,?],si64>) -> !torch.li
     %59 = torch.prim.ListConstruct %58 : (!torch.int) -> !torch.list<int>
     return %59 : !torch.list<int>
 }
+
+// -----
+
+// CHECK-LABEL: @pytorch_dynamic_pad_export_view$prop(
+func.func @pytorch_dynamic_pad_export_view$prop(%arg0: !torch.vtensor<[?,144,?,?],f32>) -> !torch.vtensor<[4,2],si64> {
+    // CHECK: %[[I0:.*]] = torch.constant.int 0
+    // CHECK: %[[x0:.*]] = torch.aten.size.int %arg0, %[[I0]] : !torch.vtensor<[?,144,?,?],f32>, !torch.int -> !torch.int
+    // CHECK: %[[I2:.*]] = torch.constant.int 2
+    // CHECK: %[[x1:.*]] = torch.aten.size.int %arg0, %[[I2]] : !torch.vtensor<[?,144,?,?],f32>, !torch.int -> !torch.int
+    // CHECK: %[[I3:.*]] = torch.constant.int 3
+    // CHECK: %[[x2:.*]] = torch.aten.size.int %arg0, %[[I3]] : !torch.vtensor<[?,144,?,?],f32>, !torch.int -> !torch.int
+    // CHECK: %[[I144:.*]] = torch.constant.int 144
+    // CHECK: %[[I0_0:.*]] = torch.constant.int 0
+    // CHECK: %[[I0_1:.*]] = torch.constant.int 0
+    // CHECK: %[[I0_2:.*]] = torch.constant.int 0
+    // CHECK: %[[I0_3:.*]] = torch.constant.int 0
+    // CHECK: %[[x3:.*]] = torch.prim.ListConstruct %[[x0]], %[[I144]], %[[x1]], %[[x2]], %[[I0_0]], %[[I0_1]], %[[I0_2]], %[[I0_3]] : (!torch.int, !torch.int, !torch.int, !torch.int, !torch.int, !torch.int, !torch.int, !torch.int) -> !torch.list<int>
+    // CHECK: %[[none:.*]] = torch.constant.none
+    // CHECK: %[[false:.*]] = torch.constant.bool false
+    // CHECK: %[[x4:.*]] = torch.aten.tensor %[[x3]], %[[none]], %[[none]], %[[false]] : !torch.list<int>, !torch.none, !torch.none, !torch.bool -> !torch.vtensor<[4,2],si64>
+    // CHECK: return %[[x4]] : !torch.vtensor<[4,2],si64>
+    %0 = torch.vtensor.literal(dense<0> : tensor<4xsi64>) : !torch.vtensor<[4],si64>
+    %1 = torch.vtensor.literal(dense<1> : tensor<1xsi64>) : !torch.vtensor<[1],si64>
+    %2 = torch.vtensor.literal(dense<0> : tensor<1xsi64>) : !torch.vtensor<[1],si64>
+    %int-9223372036854775807 = torch.constant.int -9223372036854775807
+    %int-1 = torch.constant.int -1
+    %int2 = torch.constant.int 2
+    %int1 = torch.constant.int 1
+    %int0 = torch.constant.int 0
+    %3 = torch.aten._shape_as_tensor %arg0 : !torch.vtensor<[?,144,?,?],f32> -> !torch.vtensor<[4],si64>
+    %4 = torch.prim.ListConstruct %3, %0 : (!torch.vtensor<[4],si64>, !torch.vtensor<[4],si64>) -> !torch.list<vtensor>
+    %5 = torch.aten.cat %4, %int0 : !torch.list<vtensor>, !torch.int -> !torch.vtensor<[8],si64>
+    %6 = torch.prim.ListConstruct %int-1, %int2 : (!torch.int, !torch.int) -> !torch.list<int>
+    %7 = torch.aten.view %5, %6 : !torch.vtensor<[8],si64>, !torch.list<int> -> !torch.vtensor<[4,2],si64>
+    %8 = torch.aten.slice.Tensor %7, %int0, %int-1, %int-9223372036854775807, %int-1 : !torch.vtensor<[4,2],si64>, !torch.int, !torch.int, !torch.int, !torch.int -> !torch.vtensor<[4,2],si64>
+    %9 = torch.aten.transpose.int %8, %int0, %int1 : !torch.vtensor<[4,2],si64>, !torch.int, !torch.int -> !torch.vtensor<[2,4],si64>
```

Comment:
It might be useful to add some smaller unit tests for the future if anything breaks. Especially for the transposes, since the `torch.aten.transpose`s here only test a small subset of the logic

Reply:
Yeah, because of the anchor condition, the lit tests are annoying to write, but I totally agree. I'll add a few more tests, specifically for transpose and slice.

Reply:
Thank you. Also, nice cleanup for the transpose pattern, much more readable 😄